### ISSUE TITLE
[RELAX/REBLUR] Dealing with "dancing" noise (fireflies)

### ISSUE THREAD

**Issue by LeonBrands** [NVIDIAGameWorks/RayTracingDenoiser#50]
Unrelated to my other post at #46, we're also dealing with another issue that we'd hope to improve;

We use RTXDI together with NRD, and one of the problems that we've been running into is that certain clusters of fireflies result in boiling, that seems to be enlarged by NRD. This isn't necessarily NRD's fault but I'm curious about potential tips on reducing this noise. Video here: https://tc3-img-dev.s3.eu-central-1.amazonaws.com/b1cd9d461e9cb8ffb37ffd5e253654474fda5312818aa453971029aaf54e5dd3

**Comment by dzhdanNV** [comment]
It's more a problem of RTXDI. Both RELAX and REBLUR have been designed to work with radiance. But RTXDI produces values which are `radiance / probability`. Such values converge to the ground truth if there is no spatial filtering and any kind of reprojection, if a denoiser is in action it blurs out "wrong" energy and redistributes it between neighboring pixels (if spatial stoppers allow). In general "probability" in RTXDI is sane, but there can be "high energy fireflies".

`enableAntiFirefly = true` helps in many cases, but in exchange of darkening the image a bit.

TL/DR:
- it's an area for improvement on RTXDI side
- try to filter out on the app side (you know intensity ranges)

**Comment by LeonBrands** [comment]
I see! Thank you for your help
